server/src/main/java/org/opensearch/index/translog/TranslogReader.java

```diff
@@ -118,7 +129,18 @@ public Long getCheckpointChecksum() {
     public static TranslogReader open(final FileChannel channel, final Path path, final Checkpoint checkpoint, final String translogUUID)
         throws IOException {
         final TranslogHeader header = TranslogHeader.read(translogUUID, path, channel);
-        return new TranslogReader(checkpoint, channel, path, header, null);
+
+        // When we open a reader to Translog from a path, we want to fetch the checksum
+        // as that would be needed later on while creating the metadata map for
+        // generation to checksum.
+        Long translogChecksum = null;
```

Comment:
```suggestion
        Long translogChecksum;
```

Reply:
I would prefer explicit initialisation of the variable especially since we are initialising `hasFooter` with `false`. It ensures that code is consistent and overt.